timelock-corruption-detection/src/test/java/com/palantir/timelock/history/PaxosLogHistoryProviderTest.java

```diff
@@ -0,0 +1,237 @@
+                            AcceptorUseCase.createAcceptorUseCase(useCase).value())),
+                    createLearnerLog(ImmutableNamespaceAndUseCase.of(client,
+                            LearnerUseCase.createLearnerUseCase(useCase).value())),
+                    1, idx);
+
+            historyQueries.add(HistoryQuery.of(ImmutableNamespaceAndUseCase.of(client, useCase), -1));
+            namespaceAndUseCases.add(ImmutableNamespaceAndUseCase.of(client, useCase));
+        });
+
+        List<LogsForNamespaceAndUseCase> remoteHistory
+                = HistoryLoaderAndTransformer.getLogsForHistoryQueries(history, historyQueries);
+        when(remote.getPaxosHistory(any(), any())).thenReturn(PaxosHistoryOnRemote.of(remoteHistory));
+
+        List<CompletePaxosHistoryForNamespaceAndUseCase> completeHistory = paxosLogHistoryProvider.getHistory();
+        assertThat(completeHistory.size()).isEqualTo(100);
+
+        completeHistory.forEach(historyForNamespaceAndUseCase -> {
+            Client client = historyForNamespaceAndUseCase.namespace();
+            String useCase = historyForNamespaceAndUseCase.useCase();
+
+            assertSanityOfFetchedRecords(
+                    historyForNamespaceAndUseCase,
+                    client,
+                    useCase,
+                    getIntegerValueOfClient(historyForNamespaceAndUseCase));
+
+            // to assert that history for all clients was fetched
+            assertThat(namespaceAndUseCases.remove(ImmutableNamespaceAndUseCase.of(client, useCase))).isTrue();
+
+        });
+        assertThat(namespaceAndUseCases).isEmpty();
+    }
+
+    // utils
+    private PaxosStateLog<PaxosValue> createLearnerLog(NamespaceAndUseCase namespaceAndUseCase) {
+        return SqlitePaxosStateLog.create(namespaceAndUseCase, dataSource);
+    }
+
+    private PaxosStateLog<PaxosAcceptorState> createAcceptorLog(NamespaceAndUseCase namespaceAndUseCase) {
+        return SqlitePaxosStateLog.create(namespaceAndUseCase, dataSource);
+    }
+
+    private void assertSanityOfFetchedRecords(
+            CompletePaxosHistoryForNamespaceAndUseCase historyForNamespaceAndUseCase,
+            Client client,
+            String useCase,
+            int numberOfLogs) {
+
+        assertThat(historyForNamespaceAndUseCase.useCase()).isEqualTo(useCase);
+        assertThat(historyForNamespaceAndUseCase.namespace()).isEqualTo(client);
+
+        List<ConsolidatedLearnerAndAcceptorRecord> localAndRemoteLearnerAndAcceptorRecords
+                = historyForNamespaceAndUseCase.localAndRemoteLearnerAndAcceptorRecords();
+
+        assertThat(localAndRemoteLearnerAndAcceptorRecords.size()).isEqualTo(2); // there is one local and one remote
+
+        assertThat(localAndRemoteLearnerAndAcceptorRecords.get(0).record().size())
+                .isEqualTo(localAndRemoteLearnerAndAcceptorRecords.get(1).record().size())
+                .isEqualTo(numberOfLogs);
+    }
```

Comment:
I'd like to include some assertions on what the values written are as well: assuming that they've gone through SerializationUtils they have a very specific format, which you should make part of the contract of that test method.